### PR TITLE
speech: move samples from ruby-docs-samples

### PR DESCRIPTION
**Pull request by frankyn**
- [x] move samples from ruby-docs-samples for speech
- [x] update samples to use Speech 1.0
- [x] update rspec tests to use minitest
- [x] tests pass
- [x] rubocop passes

### CODE REVIEW

File: google-cloud-speech/samples/speech_samples.rb [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+  bytes_sent     = 0
+  chunk_size     = 32_000
+
+  input_stream = Gapic::StreamInput.new
+  output_stream = speech.streaming_recognize input_stream
+
+  config = { config: { encoding:                 :LINEAR16,
+             sample_rate_hertz:        16_000,
+             language_code:            "en-US",
+             enable_word_time_offsets: true }
```
Comment (dazuma) [inline google-cloud-speech/samples/speech_samples.rb:226]: Fix the indentation (these three lines should line up with `encoding:`)

File: google-cloud-speech/samples/speech_samples.rb [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+  results = operation.response.results
+
+  alternatives = results.first.alternatives
+  alternatives.each do |alternative|
+    puts "Transcription: #{alternative.transcript}"
+  end
+  # [END speech_transcribe_async_gcs]
+end
+
+# rubocop:disable Metrics/AbcSize
```
Comment (frankyn) [inline google-cloud-speech/samples/speech_samples.rb:167]: @dazuma I added this to ignore rubocop, as it seemed a bit aggressive. If you don't agree I can work on improving this.

Comment (frankyn) [reply google-cloud-speech/samples/speech_samples.rb:167]: Added another ignore to the next sample as well.

File: google-cloud-speech/samples/acceptance/speech_samples_test.rb [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+
+require "google/cloud/storage"
+require_relative "helper"
+require_relative "../speech_samples.rb"
+
+describe "Google Cloud Speech API samples" do
+  parallelize_me!
+  before do
+    @bucket_name = ENV["GCLOUD_TEST_STORAGE_BUCKET"]
```
Comment (frankyn) [inline google-cloud-speech/samples/acceptance/speech_samples_test.rb:23]: @dazuma Is there a Cloud Storage bucket that's used by google-cloud-ruby tests? If not, I may remove the upload step to rely on storage-samples bucket.

File: google-cloud-speech/samples/speech_samples.rb (hunk 2 above)
Comment (dazuma) [reply google-cloud-speech/samples/speech_samples.rb:167]: I think that's fine.

**Pull request merged**